attribute_based.py

```diff
@@ -0,0 +1,133 @@
+
+# call aws rekognition for face unless info already exists from being called before
+def detect_face(filepath, info, client):
+    if filepath not in info.keys():
+        with open(filepath, 'rb') as image:
+            try:
+                response = client.detect_faces(Image={'Bytes': image.read()})
+            except botocore.exceptions.ClientError as error:
+                print(error)
+                this_image = Image.open(filepath)
+                this_image.thumbnail((500, 500))
+                this_image.save('temp.jpg')
+                with open('temp.jpg', 'rb') as im:
+                    response = client.detect_faces(Image={'Bytes': im.read()})
+            info[filepath] = response
+
+    return info
+
+#Note: attribute possible values and names are passed in with args: --attribute_values '2' --attribute_names 'male female'
+def size_and_distance(dataloader, args):
+    num_attrs = int(args.attribute_values)
+
+    sizes = [[] for i in range(num_attrs)]
+    tiny_sizes = [[] for i in range(num_attrs)]
+    no_faces = [[] for i in range(num_attrs)]
+    distances = [[] for i in range(num_attrs)]
+    img_center = np.array([.5, .5])
+
+
+    info = pickle.load(open('util_files/places_scene_info.pkl', 'rb'))
+    idx_to_scene = info['idx_to_scene']
+    idx_to_scenegroup = info['idx_to_scenegroup']
+    sceneidx_to_scenegroupidx = info['sceneidx_to_scenegroupidx']
+
+    if FACE_DETECT == 0:
+        client = boto3.client('rekognition')
+        if os.path.exists('{}_rekognitioninfo.pkl'.format(args.folder)):
+            detect_info = pickle.load(open('{}_rekognitioninfo.pkl'.format(args.folder), 'rb'))
+        else:
+            detect_info = {}
+    elif FACE_DETECT == 1:
+        cascPath = "haarcascade_frontalface_default.xml"
+        #Had to add in haarcascades location
+        faceCascade = cv2.CascadeClassifier(cv2.data.haarcascades + cascPath)
+
+    for i, (data, target) in enumerate(tqdm(dataloader)):
+
+        #We assume attribute information is in 1st index
+        attribute = target[1]
+
+        # Only look at image if there is an attribute to analyze (Note attribute require a bbox around the person or thing to analyze)
+        if len(attribute)> 1:
+            shape = list(data.size())[1:]
+            bbox = attribute[1]
+            bbox_adjust = np.array([bbox[0]*shape[1], bbox[1]*shape[1], bbox[2]*shape[0], bbox[3]*shape[0]])
+            pixel_size = (bbox_adjust[1]-bbox_adjust[0])*(bbox_adjust[3]-bbox_adjust[2])
+            size = (bbox[1]-bbox[0])*(bbox[3]-bbox[2])
+            person_center = np.array([bbox[0] + (bbox[1]/2.), bbox[2] + (bbox[3]/2.)])
+
+            distance = np.linalg.norm(person_center - img_center)
+            pic = data.data.cpu().numpy()
+
+            if FACE_DETECT == 0:
+                detect_info = detect_face(target[3], detect_info, client)
+                faceDetails = detect_info[target[3]]['FaceDetails']
+                if len(detect_info) % 20 == 0:
```

Comment:
I think these lines are from my previous debugging, please remove this and the following pickle dump